src/wallet/rpcwallet.cpp

```diff
@@ -2658,35 +2658,53 @@ UniValue listwallets(const JSONRPCRequest& request)
 
 UniValue openwallet(const JSONRPCRequest& request)
 {
-    if (request.fHelp || request.params.size() != 1)
+    if (request.fHelp || request.params.size() > 2)
         throw std::runtime_error(
-            "openwallet \"filename\"\n"
+            "openwallet \"filename\" (\"passphrase\")\n"
```

Comment:
Ugh, passwords on prompts leaves the password as plaintext in shell history etc. Can't really be fixed for HTTP-RPC I guess, but CLI could *probably* take a bool to mean "prompt me for password".